### DISCUSSION TITLE
I want to use my own logo.png but when I add it in the admin-interface page it does not show fully

### DISCUSSION THREAD

**Discussion by trexmo** [fabiocaccamo/django-admin-interface#365] [Q&A] [answered]
I want to use the logo here, but when I add it in the admin-interface theme it only shows the top left 20 pixels or so.

![Brolly-Logo](https://github.com/fabiocaccamo/django-admin-interface/assets/101439879/ea9963d6-8550-4d9a-b59b-75ea0020f3fd)

![Change-Theme-Django-administration](https://github.com/fabiocaccamo/django-admin-interface/assets/101439879/ee8571af-2120-4036-9d92-a1e7e5084269)

I have tried setting max height and width but no change.  

After loading the file does show in /admin-interface/

**Comment by merwok** [comment, marked as answer]
Height or width are not the problem: this is not a style issue, we can see your browser’s icon that shows the file could not be loaded.
If you look at the page HTML (with browser devtools: right click → inspect element), what URL is in the element `<img src="...">`?
If you open that URL in your browser, what do you get?
Is your project correctly configured for django media (uploaded) files?

**Reply by trexmo** [reply]
```
<a href="/admin/">
        <img class="logo" style="display:none;" src="/admin-interface/logo/Brolly-Logo.jpg">
        <span>Django administration</span>
 </a>
```
![dir-ss](https://github.com/fabiocaccamo/django-admin-interface/assets/101439879/afd9f3ac-b93a-4e80-8637-21f6b5adfba1)

I have not modified any of the admin-interface files.  

I added to settings.py
`# Base url to serve media files
MEDIA_URL = '/media/'

# Path where media is stored'
MEDIA_ROOT = BASE_DIR / 'admin-interface'`

So I created a /media/ folder at the root of the project and moved the Brolly-Logo.jpg into the /media/admin-interface/logo/ folder. That gave...

`"GET /media/admin-interface/logo/Brolly-Logo.jpg HTTP/1.1" 404 2318`

So how do I adjust such that new logo files uploaded via your admin-interface will display properly?

**Reply by merwok** [reply]
It’s strange that your media root is base_dir + /admin-interface; this is a general django setting, usually people name it `media` or `uploads`.

Have you read all the doc: https://docs.djangoproject.com/en/5.0/ref/settings/#media-url
and set up serving: https://docs.djangoproject.com/en/5.0/howto/static-files/#serving-uploaded-files-in-development

**Reply by trexmo** [reply]
I tried /admin-interface because I saw that before I added the settings changes, that is where your admin-interface module stored the uploaded log file.

**Reply by trexmo** [reply]
Ok starting over to avoid compounding errors.

Settings...

`
STATIC_URL = '/static/'

STATICFILES_DIRS = (
    os.path.join(BASE_DIR, 'static'),
)

STATIC_ROOT = os.path.join(BASE_DIR, 'staticfiles')

# Base url to serve media files
MEDIA_URL = '/media/'

# Path where media is stored'
MEDIA_ROOT = BASE_DIR / 'media'
`

Used admin Interface theme to upload new Brolly-Logo.jpg

![Brolly-Logo](https://github.com/fabiocaccamo/django-admin-interface/assets/101439879/8002a1ae-af43-40fe-bcfd-74472d43a13d)

![ss-908](https://github.com/fabiocaccamo/django-admin-interface/assets/101439879/bdc70138-f5e7-46ae-8819-0038c899188f)

`<img class="logo" style="display:none;" src="/media/admin-interface/logo/Brolly-Logo.jpg">`
 Still getting...
` "GET /media/admin-interface/logo/Brolly-Logo.jpg HTTP/1.1" 404 2318`

What am I missing?

**Reply by trexmo** [reply]
Added...

`'django.template.context_processors.media',`

to get...

`TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
                'django.template.context_processors.media',
            ],
        },
    },
]`

no change still getting...

`GET /media/admin-interface/logo/Brolly-Logo.jpg HTTP/1.1" 404 2318`

**Reply by trexmo** [reply]
Tried...

`

from django.contrib import admin
from django.urls import include, path
from django.conf import settings
from django.conf.urls.static import static

urlpatterns = [
    path("brollysaas/", include("brollysaas.urls")),
    path("admin/", admin.site.urls),
    path("media/", static(settings.MEDIA_URL,document_root=settings.MEDIA_ROOT))
]`
from the docs but 

`venv\Lib\site-packages\django\urls\conf.py", line 71, in _path  
    urlconf_module, app_name, namespace = view
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: not enough values to unpack (expected 3, got 1)`

**Comment by trexmo** [comment]
Ok the problem was in my urls.py.  Here is the working version.

`
from django.contrib import admin
from django.urls import include, path
from django.conf import settings
from django.conf.urls.static import static

urlpatterns = [
    path("brollysaas/", include("brollysaas.urls")),
    path("admin/", admin.site.urls),

]

if settings.DEBUG:
    urlpatterns += static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)`

Although I saw that in the current version of Django if debug is not needed.  This works anyway.
